### ISSUE TITLE
Request: CLI option to exit with code 0 when there are warnings but no errors

### ISSUE THREAD

**Issue by aviflax** [borkdude/clj-kondo#897]
**Is your feature request related to a problem? Please describe.**

I’m using `clj-kondo --lint` in a git pre-commit hook. In general, I don’t want to commit changes when there are lint errors. However, I often do want to commit changes when there are lint _warnings_.  I’ve tried adding some logic to my shell script that I’m using for my pre-commit hook, but shell scripting is beyond me, I’m just not smart enough to pull it off.

**Describe the solution you'd like**

I’d love a command-line switch/option that would change the exit behavior, such that when there are warnings but no errors, the tool will exit with the code 0.

**Describe alternatives you've considered**

🤔  I suppose maybe I’ll try re-writing my pre-commit hook script using Babashka.

----

Thanks!

**Comment by borkdude** [comment]
@aviflax I think clj-kondo doesn't have to change. Just use a `--config` and set the levels of the linters you don't want to disturbed about to `:level :info` or `:level :off`.

**Comment by aviflax** [comment]
Ah, good point — I’ll try that. Thanks!

**Comment by borkdude** [comment]
@aviflax It's also possible to adapt this script here:

https://github.com/borkdude/clj-kondo/blob/master/doc/ci-integration.md#pre-commit-hook

Just check if the exit code isn't equal to 3.

**Comment by aviflax** [comment]
Thanks! I might give that a try, but for now changing the level of certain linters is working great for me.

**Comment by bfontaine** [comment]
FYI I solved this issue with the following code:

```sh
clj-kondo --lint src || [ "$?" -eq "2" ]
```

It succeeds if the exit code of `clj-kondo` is 0 (no warnings and no errors) or 2 (no errors but some warnings).
